### ISSUE TITLE
[bug]: Cookies not sent with GET

### ISSUE THREAD

**Issue by dariaag** [hoppscotch/hoppscotch#4013]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I add cookies to header, it is not sent, since session user is None. Identical request through Postman returns an active session with user. 

### Steps to reproduce

I am using axum-login and tower sessions, which uses cookies to verify session, but I assume it is reproducible with any setup that requires cookies. 

### Environment

Production

### Version

Local

**Comment by chuck1in** [comment]
There seem to be a lot of problems with cookies. I use a mac desktop app, and I need to manually set cookies before they can be sent automatically. And the set cookie domain disappears after a period of time. I don't know if it is caused by expired cookies or other reasons.

**Comment by dariaag** [comment]
yes, seems like it. for me manual ones also don't work on hopscotch mac app, but httpie and postman are fine, so it's not my app for sure.

**Comment by nm17** [comment]
Can confirm, it doesn't work for whatever reason

**Comment by MexHigh** [comment]
I also have this problem. It seems like a `Set-Cookie` response header does not store the cookie in Hoppscotchs cookie manager

**Comment by jamesgeorge007** [comment]
Quick update: fixed in #6416, scheduled for the upcoming release. The desktop app now attaches stored cookies to outgoing requests.

Apologies for the long wait here.

**Comment by jamesgeorge007** [comment]
Closing since this is now addressed in [v2026.6.0](https://github.com/hoppscotch/hoppscotch/releases/tag/2026.6.0).
